static/app/components/onboarding/gettingStartedDoc/onboardingLayout.tsx

```diff
@@ -0,0 +1,174 @@
+import {Fragment, useMemo} from 'react';
+import styled from '@emotion/styled';
+
+import HookOrDefault from 'sentry/components/hookOrDefault';
+import ExternalLink from 'sentry/components/links/externalLink';
+import List from 'sentry/components/list';
+import ListItem from 'sentry/components/list/listItem';
+import {AuthTokenGeneratorProvider} from 'sentry/components/onboarding/gettingStartedDoc/authTokenGenerator';
+import {Step} from 'sentry/components/onboarding/gettingStartedDoc/step';
+import {Docs, DocsParams} from 'sentry/components/onboarding/gettingStartedDoc/types';
+import {useSourcePackageRegistries} from 'sentry/components/onboarding/gettingStartedDoc/useSourcePackageRegistries';
+import {
+  PlatformOptionsControl,
+  useUrlPlatformOptions,
+} from 'sentry/components/onboarding/platformOptionsControl';
+import {
+  ProductSelection,
+  ProductSolution,
+} from 'sentry/components/onboarding/productSelection';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {PlatformKey, Project} from 'sentry/types';
+import useOrganization from 'sentry/utils/useOrganization';
+
+const ProductSelectionAvailabilityHook = HookOrDefault({
+  hookName: 'component:product-selection-availability',
+  defaultComponent: ProductSelection,
+});
+
+export type OnboardingLayoutProps = {
+  docsConfig: Docs<any>;
+  dsn: string;
+  platformKey: PlatformKey;
+  projectId: Project['id'];
+  projectSlug: Project['slug'];
+  activeProductSelection?: ProductSolution[];
+  newOrg?: boolean;
+};
+
+const EMPTY_ARRAY: never[] = [];
+
+export function OnboardingLayout({
+  docsConfig,
+  dsn,
+  platformKey,
+  projectId,
+  projectSlug,
+  activeProductSelection = EMPTY_ARRAY,
+  newOrg,
+}: OnboardingLayoutProps) {
+  const organization = useOrganization();
+  const {isLoading: isLoadingRegistry, data: registryData} =
+    useSourcePackageRegistries(organization);
+  const selectedOptions = useUrlPlatformOptions(docsConfig.platformOptions);
+
+  const {platformOptions} = docsConfig;
+
+  const {introduction, steps, nextSteps} = useMemo(() => {
+    const {onboarding} = docsConfig;
+
+    const docParams: DocsParams<any> = {
+      dsn,
+      organization,
+      platformKey,
+      projectId,
+      projectSlug,
+      hasPerformance: activeProductSelection.includes(
+        ProductSolution.PERFORMANCE_MONITORING
+      ),
+      hasProfiling: activeProductSelection.includes(ProductSolution.PROFILING),
+      hasReplay: activeProductSelection.includes(ProductSolution.SESSION_REPLAY),
```

Comment:
nit: has* is often tied to a feature flag check - should we find a different name for it?

Reply:
hmm, maybe `showProfiling`?
Could be mistaken for a function but it is the only other concise name I can come up with. 🤔

Reply:
yep, that could work, maybe also`isProfilingSelected`? no strong preference